bug: 🐞 aggsender sanity check for PP -> FEP fails and reports error  (`v0.5.1`)

# Bug Report

## Description

This error produced by  a sanity check for the upgrade path from PP to FEP. It checks that between last certificate (PP) and new one (FEP) there are no relevant bridge events.

The problem is that check what is beetween **last settled certificate blockNumber** and **SC.StartingBlockNumber**. The code get the GAP beetween both and checks the number of claims / bridges.

Apply to version: `v0.5.1`

## To Reproduce
Steps to reproduce the behavior (FEP):

1. Configure environment: set rollup smart contract StartingBlockNumber to current L2 block number 
1. Start aggkit and send 1 certificate
2. Generate some bridges/claims
3. Stop aggkit
4. Start aggkit



## Screenshots

```
{"level":"info","ts":1757387949.1152453,"caller":"query/bridge_query.go:122","msg":"bridgeDataQuerier - finished waiting for L2 syncer to catch up to block: 966650","pid":1,"version":"v0.5.1","module":"aggsender"}
{"level":"panic","ts":1757387949.1167662,"caller":"aggsender/aggsender.go:152","msg":"error checking flow Initial Status: aggchainProverFlow - error verifying block range gaps on startup. Err: there are new bridges or claims in the gap FromBlock: 966651, ToBlock: 9684050, len(bridges)=16. len(claims)=14","pid":1,"version":"v0.5.1","module":"aggsender","stacktrace":"github.com/agglayer/aggkit/aggsender.(*AggSender).Start\n\t/app/aggsender/aggsender.go:152"}
panic: error checking flow Initial Status: aggchainProverFlow - error verifying block range gaps on startup. Err: there are new bridges or claims in the gap FromBlock: 966651, ToBlock: 9684050, len(bridges)=16. len(claims)=14

goroutine 150 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x0?, 0xc000597440?, {0x0?, 0x0?, 0xc00100f480?})
	/go/pkg/mod/go.uber.org/zap@v1.27.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00110d040, {0x0, 0x0, 0x0})
	/go/pkg/mod/go.uber.org/zap@v1.27.0/zapcore/entry.go:262 +0x24a
go.uber.org/zap.(*SugaredLogger).log(0xc0005a26e8, 0x4, {0x1a52085?, 0x41d5a5?}, {0xc000597450?, 0x1746160?, 0xc000597401?}, {0x0, 0x0, 0x0})
	/go/pkg/mod/go.uber.org/zap@v1.27.0/sugar.go:355 +0xec
go.uber.org/zap.(*SugaredLogger).Panicf(...)
	/go/pkg/mod/go.uber.org/zap@v1.27.0/sugar.go:229
github.com/agglayer/aggkit/log.(*Logger).Panicf(0xc001008240?, {0x1a52085?, 0x32aba20?}, {0xc000597450?, 0xc000ed0780?, 0x0?})
	/app/log/log.go:203 +0x45
github.com/agglayer/aggkit/aggsender.(*AggSender).Start(0xc000234600, {0x1daa008, 0x32aba20})
	/app/aggsender/aggsender.go:152 +0x1de
created by main.start in goroutine 1
	/app/cmd/run.go:125 +0x12f5
``

--- a/aggsender/flows/flow_base.go
+++ b/aggsender/flows/flow_base.go
@@ -526,6 +526,13 @@ func (f *baseFlow) VerifyBlockRangeGaps(
 	nextBlockRange := types.NewBlockRange(newFromBlock, newToBlock)
 	lastBlockRange := types.NewBlockRange(lastSettledFromBlock, lastSettledToBlock)
 
+	if lastBlockRange.Greater(nextBlockRange) {
+		// This is a strange situation, but don't need to check anything.
+		// the way of using this function is that newXXXBlock is SC.StartingBlockNumber
+		// so newXXXBlock can be a previous block
+		return nil
+	}
+
 	// case 2: is a new cert but is not contiguous to previous one
 	gap := nextBlockRange.Gap(lastBlockRange)
 	if gap.IsEmpty() {

--- a/aggsender/flows/flow_base_test.go
+++ b/aggsender/flows/flow_base_test.go
@@ -713,6 +713,18 @@ func Test_baseFlow_VerifyBlockRangeGaps(t *testing.T) {
 					Return([]bridgesync.Bridge{}, []bridgesync.Claim{}, nil)
 			},
 		},
+		{
+			name: "lastSentCertificate is Settled, and SC.StartingBlockNumber is < that last Cert, so nothing to check",
+			args: args{
+				lastSentCertificate: &types.CertificateHeader{
+					Status:    agglayertypes.Settled,
+					FromBlock: 5,
+					ToBlock:   10,
+				},
+				newFromBlock: 1,
+				newToBlock:   1,
+			},
+		},
 	}
 
 	for _, tt := range tests {

--- a/aggsender/types/block_range.go
+++ b/aggsender/types/block_range.go
@@ -64,6 +64,14 @@ func (b BlockRange) Gap(other BlockRange) BlockRange {
 	}
 }
 
+// Greater returns true if the receiver BlockRange (b) is strictly greater than the other BlockRange (other).
+// [ 10 - 50 ] > [ 1 - 9 ] = true
+// [ 10 - 50 ] > [ 5 - 15 ] = false (overlap)
+// [ 10 - 50 ] > [ 51 - 100 ] = false (not greater)
+func (b BlockRange) Greater(other BlockRange) bool {
+	return b.FromBlock > other.ToBlock
+}
+
 func getBlockMinusOne(fromBlock uint64) uint64 {
 	if fromBlock > 0 {
 		return fromBlock - 1

--- a/aggsender/types/block_range_test.go
+++ b/aggsender/types/block_range_test.go
@@ -154,3 +154,91 @@ func TestBlockRange_IsEmpty(t *testing.T) {
 		})
 	}
 }
+func TestBlockRange_Greater(t *testing.T) {
+	tests := []struct {
+		name     string
+		a        BlockRange
+		b        BlockRange
+		expected bool
+	}{
+		{
+			name:     "a strictly greater than b",
+			a:        NewBlockRange(10, 20),
+			b:        NewBlockRange(1, 9),
+			expected: true,
+		},
+		{
+			name:     "a overlaps b",
+			a:        NewBlockRange(10, 20),
+			b:        NewBlockRange(15, 25),
+			expected: false,
+		},
+		{
+			name:     "a adjacent to b",
+			a:        NewBlockRange(11, 20),
+			b:        NewBlockRange(1, 10),
+			expected: true,
+		},
+		{
+			name:     "a not greater than b",
+			a:        NewBlockRange(1, 5),
+			b:        NewBlockRange(6, 10),
+			expected: false,
+		},
+		{
+			name:     "identical ranges",
+			a:        NewBlockRange(5, 10),
+			b:        NewBlockRange(5, 10),
+			expected: false,
+		},
+		{
+			name:     "a starts after b ends by 1",
+			a:        NewBlockRange(11, 15),
+			b:        NewBlockRange(1, 10),
+			expected: true,
+		},
+		{
+			name:     "a starts at b end",
+			a:        NewBlockRange(10, 15),
+			b:        NewBlockRange(1, 10),
+			expected: false,
+		},
+		{
+			name:     "empty a, non-empty b",
+			a:        NewBlockRange(0, 0),
+			b:        NewBlockRange(1, 10),
+			expected: false,
+		},
+		{
+			name:     "non-empty a, empty b",
+			a:        NewBlockRange(5, 10),
+			b:        NewBlockRange(0, 0),
+			expected: true,
+		},
+		{
+			name:     "both empty",
+			a:        NewBlockRange(0, 0),
+			b:        NewBlockRange(0, 0),
+			expected: false,
+		},
+		{
+			name:     "invalid a (from > to)",
+			a:        NewBlockRange(10, 5),
+			b:        NewBlockRange(1, 4),
+			expected: true,
+		},
+		{
+			name:     "invalid b (from > to)",
+			a:        NewBlockRange(5, 10),
+			b:        NewBlockRange(10, 5),
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.a.Greater(tt.b)
+			require.Equal(t, tt.expected, got, "Greater() for %s: expected %v, got %v", tt.name, tt.expected, got)
+		})
+	}
+}

--- a/go.mod
+++ b/go.mod
@@ -120,7 +120,7 @@ require (
 	github.com/go-playground/locales v0.14.1 // indirect
 	github.com/go-playground/universal-translator v0.18.1 // indirect
 	github.com/go-playground/validator/v10 v10.26.0 // indirect
-	github.com/go-viper/mapstructure/v2 v2.3.0 // indirect
+	github.com/go-viper/mapstructure/v2 v2.4.0 // indirect
 	github.com/goccy/go-json v0.10.5 // indirect
 	github.com/gofrs/flock v0.12.1 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect

